### DISCUSSION TITLE
Does not support requestanimationframe, which is equivalent to setinterval. When the 3D scene is not displayed, it still occupies CPU and GPU

### DISCUSSION THREAD

**Discussion by 0611163** [cefsharp/CefSharp#4041] [General]
CefSharp. WinForms v86. 0.241, does not support requestanimationframe, which is equivalent to setinterval. When the 3D scene is not displayed, it still occupies CPU and GPU

CefSharp.WinForms v86.0.241, 不支持requestAnimationFrame, requestAnimationFrame相当于setInterval, 三维场景不显示时依然占用CPU和GPU

- **What architecture x86 or x64?**
    x86
	
- **What version of .Net?**
    .Net 4.x
    
- **On what operating system?**
    Win10

- **Are you using `WinForms`, `WPF` or `OffScreen`?**
    WinForms

**Comment by amaitland** [comment]
> CefSharp. WinForms v86. 0.241, does not support requestanimationframe, which is equivalent to setinterval. When the 3D scene is not displayed, it still occupies CPU and GPU

Please don't open bugs for old versions.

Version `99` is the current supported version. 

There is an upstream discussion about high `CPU` usage at https://magpcss.org/ceforum/viewtopic.php?f=6&t=18758 I'd suggest you join in the conversation there.

**Comment by 0611163** [comment]
Thank you for your reply.&nbsp;


I have another problem, which is cefsharp v86.0.241 in WinForm version&nbsp; does not use separate graphics card. I have a computer with dual graphics cards. I set up high performance through win10.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Alex ***@***.***&gt;; 
发送时间: 2022年3月17日(星期四) 中午11:30
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [cefsharp/CefSharp] Does not support requestanimationframe, which is equivalent to setinterval. When the 3D scene is not displayed, it still occupies CPU and GPU (Discussion #4041)





  
CefSharp. WinForms v86. 0.241, does not support requestanimationframe, which is equivalent to setinterval. When the 3D scene is not displayed, it still occupies CPU and GPU
  
Please don't open bugs for old versions.
 
Version 99 is the current supported version.
 
There is an upstream discussion about high CPU usage at https://magpcss.org/ceforum/viewtopic.php?f=6&amp;t=18758 I'd suggest you join in the conversation there.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by amaitland** [reply]
> I have another problem, which is cefsharp v86.0.241 in WinForm version&nbsp; does not use separate graphics card. I have a computer with dual graphics cards. I set up high performance through win10.

This is not a supported version.

**Reply by amaitland** [reply]
> does not use separate graphics card. I have a computer with dual graphics cards

`Chrome` doesn't support selecting a `GPU` directly as as such `CefSharp` doesn't have a feature. A quick search yields https://support.google.com/chrome/thread/77636044/chrome-to-use-specific-gpu?hl=en

Ultimately you'll need to do your own research into this. I don't have dual `GPU's`

**Reply by amaitland** [reply]
Only other comment I'll make is that CefSharp.BrowserSubProcess.exe is used as the GPU process by default.

**Comment by 0611163** [comment]
Thank you for your reply. 


I have another problem, which is cefsharp v86.0.241 in WinForm version&nbsp; does not use separate graphics card. I have a computer with dual graphics cards. I set up high performance through win10.





------------------ 原始邮件 ------------------
发件人: "Alex ***@***.***&gt;; 
发送时间: 2022年3月17日(星期四) 中午11:30
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [cefsharp/CefSharp] Does not support requestanimationframe, which is equivalent to setinterval. When the 3D scene is not displayed, it still occupies CPU and GPU (Discussion #4041)





  
CefSharp. WinForms v86. 0.241, does not support requestanimationframe, which is equivalent to setinterval. When the 3D scene is not displayed, it still occupies CPU and GPU
  
Please don't open bugs for old versions.
 
Version 99 is the current supported version.
 
There is an upstream discussion about high CPU usage at https://magpcss.org/ceforum/viewtopic.php?f=6&amp;t=18758 I'd suggest you join in the conversation there.
 
—
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by 0611163** [comment]
thank you. The solution you gave is useful. The problem is solved. thank you.




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Alex ***@***.***&gt;; 
发送时间: 2022年3月17日(星期四) 下午2:08
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [cefsharp/CefSharp] Does not support requestanimationframe, which is equivalent to setinterval. When the 3D scene is not displayed, it still occupies CPU and GPU (Discussion #4041)





 
Only other comment I'll make is that CefSharp.BrowserSubProcess.exe is used as the GPU process by default.
 
&mdash;
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.
